src/elasticsearch/index.ts

```diff
@@ -1,166 +1,440 @@
 import getFunctionScores from './score'
 import getMultiMatchConfig from './multimatch'
 import getBoosts from './boost'
-import getMapping from './mapping'
 import cloneDeep from 'clone-deep'
 import SearchQuery from '../types/SearchQuery'
+import AppliedFilter from '../types/AppliedFilter'
 import ElasticsearchQueryConfig from './types/ElasticsearchQueryConfig'
 
-function processNestedFieldFilter (attribute: string, value: any) {
-  let processedFilter = {
-    'attribute': attribute,
-    'value': value
-  };
-  let filterAttributeKeys = Object.keys(value);
-  for (let filterAttributeKey of filterAttributeKeys) {
-    if (value[filterAttributeKey] && !Array.isArray(value[filterAttributeKey]) && typeof value[filterAttributeKey] === 'object') {
-      processedFilter = processNestedFieldFilter(attribute + '.' + filterAttributeKey, value[filterAttributeKey]);
-    }
-  }
-  return processedFilter;
+export interface FilterOptions {
+  operator: string
+  attribute: string
+  value: any,
+  queryChain?: any
+}
+
+export interface FilterInterface {
```

Comment:
Can you please describe all those methods as much as possible? When someone will import this interface to create customFilters then he will know what he need to do

Reply:
I will using JSDocs. But actually I also would write a recipe in the VSF docs as well.

Reply:
If described the main functions and interfaces as much as possible.